### PR TITLE
21696-fix broken code coverage uploads in CIs

### PR DESCRIPTION
**Pull request by eason-pan-bc**
*Issue #:* /bcgov/entity#21696

*Description of changes:*
- Update Codecov version to use v4, and add token to them
- Add Codecov section for Furnishings Job CI and Involuntary Dissolutions Job CI

Fixed and verified for legal-api before, PRs:
Add token and using v4: https://github.com/bcgov/lear/pull/2759
Verify it works for legal-api: https://github.com/bcgov/lear/pull/2760

Will create another test PR with other CIs after this PR gets merged.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

**Review by JazzarKarim — APPROVED** [review]
LGTM 👍

**Review by AimeeGao — APPROVED** [review]
Nice! looking forward to the other CIs being addressed.

File: .github/workflows/furnishings-ci.yml [hunk 1]
```diff
@@ -82,6 +82,14 @@ jobs:
         id: test
         run: |
           make test
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v4
+        with:
+          file: ./jobs/furnishings/coverage.xml
```
Comment (chenhongjing) [inline .github/workflows/furnishings-ci.yml:88]: Seems furnishings & involuntary dissolutions jobs don't have this file in place. Can you add it?

**Comment by chenhongjing** [comment]
Others look good!

File: .github/workflows/furnishings-ci.yml (hunk 1 above)
Comment (eason-pan-bc) [reply .github/workflows/furnishings-ci.yml:88]: Thank you for mentioning this. However, coverage.xml is in the gitignore, so they won't be push to repos anyway 
image

This inspired me to check the `codecov.yaml`. @argush3 will that be a good idea to add flags under coverage and adding  paths for Entity BN, Entity Digital Credentials, Furnishings Job, Involuntary Dissolutions Job to `codecov.yaml` file?
image

Comment (argush3) [reply .github/workflows/furnishings-ci.yml:88]: yes, I think that woudl be a good idea to add these flags to codegov.yaml.  I would remove datareset too

**Comment by eason-pan-bc** [comment]
Added Entity BN, Entity Digital Credentials, Furnishings Job, Involuntary Dissolutions flags to codecov.yaml, and removed datareset flag from it.

**Review by argush3 — APPROVED** [review]

**Review by chenhongjing — APPROVED** [review]

**Pull request merged**